source/Missions/Services/Network/MissionClient.cs

```diff
@@ -95,7 +95,7 @@ private void Handle_JoinInfo(MessagePayload<MissionJoinInfo> payload)
                 joinInfo.CharacterObject.IsPlayerCharacter ? "Player" : "Agent",
                 joinInfo.CharacterObject.Name, newAgentId, netPeer.EndPoint);
             // TODO remove test code
-            Agent newAgent = MissionTestGameManager.SpawnAgent(startingPos, joinInfo.CharacterObject);
+            Agent newAgent = ArenaTestGameManager.SpawnAgent(startingPos, joinInfo.CharacterObject);
```

Comment:
Create a IMissionGameManager interface with common methods. Update MissionClient to use IMissionGameManager as a dependency and make this line generic.

This won't work with taverns which we still want